### PR TITLE
Adds Strolling Mode

### PR DESCRIPTION
**Pull request by aej11a**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -26,12 +26,19 @@ module.exports = {
 	overrides: [
 		{
 			files: ["*.ts", "*.tsx"],
-			extends: ["plugin:@typescript-eslint/recommended-requiring-type-checking"],
+			//extends: ["plugin:@typescript-eslint/recommended-requiring-type-checking"],
```
Comment (carpenterd777) [inline .eslintrc.js:29]: why did you disable this?

File: .eslintrc.js [hunk 2]
```diff
@@ -26,12 +26,19 @@ module.exports = {
...
 			parserOptions: {
 				project: ["./tsconfig.json"],
 			},
 			rules: {
 				"@typescript-eslint/switch-exhaustiveness-check": "error",
+				"@typescript-eslint/no-floating-promises": [
+					"error",
+					{
+						ignoreVoid: true, // this is the default
+						ignoreIIFE: true, // this is different from the default
```
Comment (carpenterd777) [inline .eslintrc.js:39]: I like this rule change

File: src/components/Button.tsx [hunk 3]
```diff
@@ -13,12 +13,20 @@ export interface ButtonProps {
 	style?: object; // may be unsafe, but this is the type provided
 	// by Stylesheet documentation:
 	// https://reactnative.dev/docs/stylesheet#compose
+	children?: JSX.Element;
```
Comment (carpenterd777) [inline src/components/Button.tsx:16]:
```suggestion
	children?: Array<JSX.Element>;
```

I think I remember needing this to be an array for a component I made while working but I could be misremembering

File: src/components/Button.tsx [hunk 4]
```diff
@@ -33,24 +41,24 @@ const Button = ({ image, text, accessibilityLabel, onPress, style }: ButtonProps
 		},
 	});
 
-	// chooses one to display, favoring bottom most prop
+	// chooses one to display, favoring top most prop
```
Comment (carpenterd777) [inline src/components/Button.tsx:44]: yeah, that probably makes more sense that what I did

File: src/components/NearbyPlaces/useLocation.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from "react";
+import * as Location from "expo-location";
+
+export const useLocation = () => {
```
Comment (carpenterd777) [inline src/components/NearbyPlaces/useLocation.ts:4]:
```suggestion
export const useLocation = () : {Location.LocationObject, string}=> {
```

**Review by carpenterd777 — CHANGES_REQUESTED** [review]
I'm getting this strange error

image

This is strange only because my location services are very clearly enabled, right down to my phone having logged a recent location request from Expo Go. I'll put a screenshot in a comment under this. 

I like the folder structure changes, it bothers me that by default expo projects have no `/src` lol. But I think maybe `/utils` should go in there too. I guess `/tests` can stay outside, and `App.tsx` too.

**Comment by elenirotsides** [comment]
After I accepted the location request, I get stuck in an infinite "Loading..." screen. Not sure if this is something I caused or an oversight, but thought I'd document this:

(This is a screen recording of what I can see. Of course, this is after the initial run of the app where it asked permission to see my location)

https://user-images.githubusercontent.com/43190242/135951575-260ef67d-f4db-4bf0-8b19-4c586b3c2c0e.mov

File: src/components/Button.tsx (hunk 3 above)
Comment (aej11a) [reply src/components/Button.tsx:16]: Actually technically the right thing to do is use React.FC<{other props}> and it'll add in children with the correct type, i guess I was just being lazy and didn't want to change what was already there but i'll fix it

**Comment by aej11a** [comment]
Let's debug these issues in person today, seems device specific

**Comment by carpenterd777** [comment]
> 
> 
> I'm getting this strange error
> 
> image
> 
> This is strange only because my location services are very clearly enabled, right down to my phone having logged a recent location request from Expo Go. I'll put a screenshot in a comment under this.
> 
> I like the folder structure changes, it bothers me that by default expo projects have no `/src` lol. But I think maybe `/utils` should go in there too. I guess `/tests` can stay outside, and `App.tsx` too.

As of 10-05-21, we will merge this in with the Android permission error and open a new issue for it.

**Review by carpenterd777 — APPROVED** [review]

**Pull request merged**